common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/Indexer.scala

```diff
@@ -25,14 +24,53 @@ trait Indexable[T] {
   def version(document: T): Int
 }
 
+/**
+  * Versioning:
+  *
+  * When the merger makes the decision to merge some works, it modifies
+  * the content of the affected works. Despite the content of these works
+  * being modified, their version remains the same. However, the merger
+  * attaches the number of merged sources (`nMergedSources`) to the target
+  * work and may choose to redirect some of the source works.
```

Comment:
Not a change for this PR, but I wonder if this speaks to a problem with the word "version"?

The "version" on a Work corresponds to the version of the work in the source system, yes? Maybe we should rename it to "sourceVersion" in a future PR, to emphasise everywhere that it's not counting distinct versions of the work?

(I was going to say "a la SourceIdentifier", but now I wonder – should it be a field *on* SourceIdentifier?)

Reply:
That's a very good point - it's appealing to put it on `SourceIdentifier` certainly, worth thinking about